storage/objects/generate_signed_post_policy_v4.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package objects
+
+// [START storage_generate_signed_post_policy_v4]
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"golang.org/x/oauth2/google"
+)
+
+// generateSignedPostPolicyV4 generates a signed post policy.
+func generateSignedPostPolicyV4(w io.Writer, bucket, object, serviceAccountJSON string) (*storage.PostPolicyV4, error) {
```

Comment:
This signature is a little confusing given that the body of this form manually creates a form then also returns the storage.PostPolicyV4. I'd think that perhaps we'd make it return:
a) Generated form string
b) URL string
c) error

or perhaps show the complete upload as we have in example_test.go

Reply:
The return in this case is added for integration tests. @tritone and I discussed this further and it was a tradeoff to make it easier to verify in tests that the form values are valid.

Reply:
Just to add additional context, samples in other languages return the form itself but lack integration tests to actually verify that the post will work as expected against GCS. I discussed this with Frank and we think it's best to add that integration test in other languages as well. But that work can come later.